### ISSUE TITLE
Incorrect selection after task completion (only with lua?)

### ISSUE THREAD

**Issue by smichel17** [mpcjanssen/simpletask-android#840]
**Steps to reproduce:**

- Load filter (with lua onFilter?) where completed tasks are not shown
- Select a task
- Complete the selected task

**Expected result:** Selected task is completed, remains selected and visible.
**Actual result:** Selected task is completed but a different task is selected and visible.

> <img alt="screenshot" src="https://user-images.githubusercontent.com/6563664/34307055-2e72db42-e73e-11e7-93dc-4abf5a52a95e.png" height="450">
>
> The selected task is blank because I'm using the `onDisplay` callback to show only task text, and that particular task is just a link

I could not reproduce this with a simple filter. I've only seen it in this particular filter, that uses a fair amount of lua scripts. Here's the filter:

```lua
  "> TODAY": {
    "TITLE": "> TODAY",
    "CONTEXTS": "",
    "CONTEXTSnot": false,
    "PROJECTS": "",
    "PROJECTSnot": false,
    "PRIORITIES": "",
    "PRIORITIESnot": false,
    "SORTS": "+!by_lua\n+!completed\n+!by_prio\n+!by_due_date\n-!by_threshold_date\n+!alphabetical\n+!file_order\n-!by_project\n-!by_context\n+!in_future\n-!by_creation_date\n+!by_completion_date",
    "HIDECOMPLETED": false,
    "HIDEFUTURE": true,
    "HIDELISTS": false,
    "HIDETAGS": true,
    "HIDECREATEDATE": true,
    "HIDEHIDDEN": false,
    "CREATEISTHRESHOLD": false,
    "LUASCRIPT": "function onFilter ( t, f, e )\n  return todoToday(f)\nend\n\nfunction onGroup(t,f,e)\n  return group(f)\nend\n\nfunction onSort(t,f,e)\n  return SORT[group(f)]\nend\n\nfunction onDisplay(t,f,e)\n  return bare(f)\nend\n",
    "USE_SCRIPT": true,
    "LUASCRIPT_TEST_TASK": "(B) 2017-11-26 +k test task",
    "query": ""
  },
```
and the corresponding lua config to make it work:

```
---------------------------------------
-- External (use in filters)
---------------------------------------

NOTTASK="error: not a task"
DONE="done"
TODAY="today"
STRETCH="stretch"
SOON="soon"
LATER="later"
SOMEDAY="someday/maybe"

SORT={ [TODAY]="1", [STRETCH]="2", [SOON]="3", [LATER]="4", [SOMEDAY]="5", [DONE]="6" }

function todoToday(f)
  g=group(f)
  return g==TODAY or g==STRETCH
    or (g==DONE and doneToday(f))
end

function todoList(f)
  g=group(f)
  return g==TODAY or g==STRETCH
    or g==SOON 
end

--> For onGroup

-- This is the core logic that everything else is based around
function group(f)
  if not isTask(f) then
    return NOTTASK
  elseif f.completed then
    return DONE
  elseif not isPlanned(f) then
    return SOMEDAY
  elseif groupToday(f) then
    return TODAY
  elseif groupStretch(f) then
    return STRETCH
  elseif inNext(WEEK, f.due) then
    return SOON
  else
    return LATER
  end
end

--> for onDisplay

function bare(f)
  return f.text
end

------------------------
-- Todo helpers
------------------------

function isPlanned(f)
  return f.priority ~= "-"
    or f.due ~= nil
end

function isTask(f)
  return next(f.tags) ~= nil
end

function isList(f)
  return next(f.tags) == nil
      and next(f.lists) ~= nil
end

function groupToday(f)
  return f.priority == "A"
    or inLast(2*DAY, f.due)
end

function doneToday(f)
  return inLast(DAY, f.completiondate)
end

stretch={ ["A"]=false, ["B"]=true }

-- If today is closer to the due than threshold date
function groupStretch(f)
  p=max(os.time()-threshold(f),DAY)
  return stretch[f.priority]
    or inNext(p, f.due)
end

-- Fallbacks if the threshold isn't set
function threshold(f)
  if (f.threshold~=nil) then
    return f.threshold
  elseif (f.createdate~=nil) then
    return f.createdate
  else
    return os.time()
  end
end

--------------------------------
-- Utils
--------------------------------

function max(a,b)
  if (a>=b) then
    return a
  else
    return b
  end
end

DAY=86400
WEEK=DAY*7

function inLast(period, time)
  now=os.time()
  return time ~= nil
      and time < now
      and time >= (now - period)
end

function inNext(period, time)
  now=os.time()
  return time ~= nil
      and time > now
      and time <= (now + period)
end
```

**Comment by mpcjanssen** [comment]
@smichel17 can you check the behavior in the latest playstore version (10.1.13) there have been a lot of filtering and Lua changes which might have affected this.

**Comment by smichel17** [comment]
Waiting on fdroid version or getting my build environment set up; I'll let you know when one of those things happens :)

**Comment by mpcjanssen** [comment]
I think I completely broke keep selection, but I am happy to be proven otherwise.

**Comment by smichel17** [comment]
Doesn't exactly help, but same bug on latest fdroid (10.1.3).

**Comment by mpcjanssen** [comment]
Wel anything before 10.1.12 is broken in some way so that's not surprising.
